components/proxy/src/main/java/com/hotels/styx/startup/CoreMetrics.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (C) 2013-2018 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.styx.startup;
+
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.jvm.BufferPoolMetricSet;
+import com.codahale.metrics.jvm.GarbageCollectorMetricSet;
+import com.codahale.metrics.jvm.MemoryUsageGaugeSet;
+import com.codahale.metrics.jvm.ThreadStatesGaugeSet;
+import com.hotels.styx.Environment;
+import com.hotels.styx.api.metrics.MetricRegistry;
+import com.hotels.styx.metrics.reporting.sets.NettyAllocatorMetricSet;
+import io.netty.buffer.PooledByteBufAllocator;
+import io.netty.buffer.UnpooledByteBufAllocator;
+import org.slf4j.Logger;
+
+import java.lang.management.RuntimeMXBean;
+import java.time.Duration;
+import java.util.Optional;
+
+import static java.lang.String.format;
+import static java.lang.management.ManagementFactory.getPlatformMBeanServer;
+import static java.lang.management.ManagementFactory.getRuntimeMXBean;
+import static org.slf4j.LoggerFactory.getLogger;
+
+/**
+ * Core metrics - JVM details, Styx version.
+ */
+public final class CoreMetrics {
+    private static final Logger LOG = getLogger(CoreMetrics.class);
+
+    private CoreMetrics() {
+    }
+
+    public static void registerCoreMetrics(Environment environment) {
```

Comment:
Instead of passing in all of the environment, pass in only the objects that are actually used: 1) buildInfo, and 2) metrics registry. It makes it easier to follow the code at the `registerCoreMetrics` call site.

Reply:
We should really have some unit tests for this class.